### DISCUSSION TITLE
Step 3: Exchange authorization code for bearer token not getting Location in header

### DISCUSSION THREAD

**Discussion by labdel** [timdorr/tesla-api#336] [General]
Hi All,
I am following up code in https://github.com/enode-engineering/tesla-oauth2/blob/main/tesla.py and I am facing some issues this week after it was working last week.

After getting the 1st Authorization in the response it gets the csrf and transaction_id, then it executes the 2nd authorization, it return 200 and "<title>Tesla SSO - Sign In</title>" - it never returns 302. Headers does not contain 
location so it can pull the code in the "https://auth.tesla.com/void/callback?code

Does anyone know what could be causing it ?

Thanks in advance - Leo.

**Comment by pgalbavy** [comment]
I got this when I didn't send the Cookie back recieved in the first step.

**Reply by labdel** [reply]
Thanks Peter, I will change the code accordingly.

Regards

**Reply by pgalbavy** [reply]
Of course right after telling you this and tidying up my own test code, I forgot to send the cookie to step 3 and got 403 Forbidden, could not work out why. So, you need it there too! Set 4, exchange token for bearer token should not need ti but I am not sure.
